execution-engine/src/wasi/wasmi.rs

```diff
@@ -1226,6 +1226,8 @@ impl ExecutionEngine for WASMIRuntimeState {
         file_name: &str,
         options: Options,
     ) -> Result<u32, FatalEngineError> {
+        self.vfs.environment_variables = options.environment_variables;
+        self.vfs.program_arguments = options.program_arguments;
```

Comment:
Could we move those two variables to somewhere else? Do they have to be in the VFS?

Reply:
I've already moved them, in a later commit. Perhaps I should rebase and squash at this point.

Reply:
I've squashed and rebased. (For some reason GPG has stopped working, though.)